core/src/main/java/com/ritense/valtimo/service/CamundaProcessService.java

```diff
@@ -560,12 +546,33 @@ private boolean isProcessDefinitionPreviouslyDeployed(
                 outputStream.close();
 
             } catch (IOException e) {
-                throw new ProcessNotDeployableException(caseDefinitionId + " and process: " + processDefinitionKey);
+                throw new ProcessNotDeployableException(caseDefinitionId + " and process: " + latestProcessDefinition.getKey());
             }
         }
         return false;
     }
 
+    public ProcessDefinition getExistingProcessForFile(
+        CaseDefinitionId caseDefinitionId,
+        BpmnModelInstance bpmnModel
+    ) {
+        String processDefinitionKey = bpmnModel.getModelElementsByType(Process.class).stream()
+            .map(Process::getId)
+            .findFirst().orElseThrow();
+
+        var query = repositoryService
+            .createProcessDefinitionQuery()
+            .processDefinitionKey(processDefinitionKey)
+            .latestVersion()
+            .active();
+        if (caseDefinitionId != null) {
+            query = query.versionTag(CAMUNDA_CASE_DEFINITION_VERSION_TAG_PREFIX + caseDefinitionId);
+        } else {
+            query = query.withoutVersionTag();
```

Comment:
tnh, I dont think this is correct since people are still allowed to use their own version tags. Is it possible to get it from the `bpmnModel`?

Reply:
fixed